keps/20190514-operator-toolkit.md

```diff
@@ -0,0 +1,53 @@
+---
+kep-number: 10
+title: KUDO Operator Toolkit
+authors:
+  - "@kensipe"
+  - "@gerred"  
+owners:
+  - "@kensipe"
+  - "@gerred"
+editor: "@kensipe"
+creation-date: 2019-10-14
+last-updated: 2019-10-14
+status: provisional
+see-also:
+  - KEP-0002
+---
+
+# KUDO Operator Toolkit
+
+## Table of Contents
+
+* [KUDO Operator Toolkit](#kudo-operator-toolkit)
+     * [Table of Contents](#table-of-contents)
+     * [Summary](#summary)
+     * [Motivation](#motivation)
+     * [Proposal](#proposal)
+        * [Goals](#goals)
+        * [Non-Goals](#non-goals)
+
+## Summary
+
+Drive KUDO to have a set of tools, frameworks and specifications for creating operators.
+
+
+## Motivation
+
+KUDO provides a way to reduce the amount of code necessary for the creation of an operator for Kubernetes. The current implementation of KUDO requires a significant amount of YAML to define an operator.   This YAML is verbose, is not reusable, and requires [kustomize](https://github.com/kubernetes-sigs/kustomize) and [mustache](https://mustache.github.io/).  We believe we can significantly improve the operator developers experience through the creation of a toolkit.
+
+
+## Proposal
+
+### Goals
```

Comment:
I think we need to look back at the Requirements in https://hackmd.io/Und9SUwzSZGmKV38LIx6cA?both.  Particular ones I think we need to ensure work:

* Extensibility of FrameworkVersion:  Adding plans, parameters, etc
* As a FrameworkDeveloper (FD) I should be able to re-use tasks/phases/steps.
* As a FD I should be able to define "DataCRDS"
* As the FD I should be able to extend the Kudo CLI for my framework.